In [1]:
import pandas as pd

In [40]:
df = pd.read_excel("2021.xlsx")

In [41]:
df.head()

Commodity or Services       Category  Lowest  Second  Third  \
0          Bread, rice and cereals  carbohydrates     1.6     1.6    1.5   
1                   Pasta products  carbohydrates     0.2     0.1    0.1   
2       Buns, cakes, biscuits etc.  carbohydrates     1.2     1.3    1.1   
3                 Pastry (savoury)  carbohydrates     0.3     0.3    0.2   
4  Beef (fresh, chilled or frozen)        protein     0.5     0.5    0.5   

   Fourth   Fifth   Sixth   Seventh  Eighth   Ninth   Highest  year  
0      1.4     1.3     1.3      1.1      1.1     1.0      0.8  2021  
1      0.1     0.1     0.1      0.1      0.1     0.1      0.1  2021  
2      1.1     0.9     0.8      0.8      0.8     0.7      0.5  2021  
3      0.2     0.2     0.2      0.2      0.2     0.2      0.1  2021  
4      0.5     0.5     0.5      0.4      0.5     0.4      0.4  2021

In [42]:
df = df.melt(id_vars = ['Commodity or Services','Category','year'],var_name = 'income_group', value_name = 'percentage_of_income')

In [5]:
df.head()

Commodity or Services       Category  year income_group  \
0          Bread, rice and cereals  carbohydrates  2021       Lowest   
1                   Pasta products  carbohydrates  2021       Lowest   
2       Buns, cakes, biscuits etc.  carbohydrates  2021       Lowest   
3                 Pastry (savoury)  carbohydrates  2021       Lowest   
4  Beef (fresh, chilled or frozen)        protein  2021       Lowest   

   percentage_of_income  
0                   1.6  
1                   0.2  
2                   1.2  
3                   0.3  
4                   0.5

In [6]:
df.to_excel('2021_1.xlsx')

## Descriptive Analysis

In [1]:
import pandas as pd
df = pd.read_excel('2021_1.xlsx')

In [44]:
df.head()

Commodity or Services Category  year income_group  \
0          Bread, rice and cereals     food  2021  Lower Class   
1                   Pasta products     food  2021  Lower Class   
2       Buns, cakes, biscuits etc.     food  2021  Lower Class   
3                 Pastry (savoury)     food  2021  Lower Class   
4  Beef (fresh, chilled or frozen)     food  2021  Lower Class   

   percentage_of_income  
0                   1.6  
1                   0.2  
2                   1.2  
3                   0.3  
4                   0.5

In [2]:
df['year'] = pd.to_datetime(df["year"], format="%Y")

In [4]:
df.groupby('income_group').describe()

percentage_of_income                                     \
                                  count      mean       std  min  25%  50%   
income_group                                                                 
Lower Class                       756.0  1.156085  2.813365  0.0  0.2  0.4   
Lower Middle Class               1008.0  1.150496  2.637259  0.0  0.2  0.4   
Upper Class                       252.0  1.142857  2.580217  0.0  0.1  0.3   
Upper Middle Class                504.0  1.160516  2.603333  0.0  0.1  0.3   

                              avg weekly income                            \
                    75%   max             count   mean  std    min    25%   
income_group                                                                
Lower Class         0.8  24.7             756.0  136.0  0.0  136.0  136.0   
Lower Middle Class  0.8  21.5            1008.0  390.0  0.0  390.0  390.0   
Upper Class         0.7  15.4             252.0  745.0  0.0  745.0  745.0   
Upper Middle Class  0.8  16.6             504.0  626.0  0.0  626.0  626.0   

                                         
                      50%    75%    max  
income_group                             
Lower Class         136.0  136.0  136.0  
Lower Middle Class  390.0  390.0  390.0  
Upper Class         745.0  745.0  745.0  
Upper Middle Class  626.0  626.0  626.0

1. There are 756 lower class households in the survey.
2. There are 1008 lower middle class households.
3. 504 upper middle class
4. 252 Upper Class households. 

In [10]:
df.groupby('Category').describe()

percentage_of_income                            \
                                         count       mean       std  min   
Category                                                                   
alcohol and intoxicants                  160.0   0.557500  0.329389  0.0   
clothings and accessories                300.0   0.493000  0.942592  0.0   
communication                             30.0   4.083333  0.716834  2.5   
education                                150.0   0.390000  0.489453  0.1   
entertainment                            650.0   0.801077  0.918286  0.0   
food                                     990.0   0.362929  0.332970  0.0   
health                                    60.0   0.393333  0.152790  0.1   
household                                 90.0  10.507778  5.014546  5.4   
miscellenous                              30.0   7.663333  0.563538  6.5   
other                                     30.0   0.553333  0.133218  0.3   
transportation                            30.0  13.090000  2.048439  8.4   

                                                        
                              25%    50%     75%   max  
Category                                                
alcohol and intoxicants     0.300   0.50   0.700   1.7  
clothings and accessories   0.100   0.20   0.600   8.8  
communication               3.650   4.30   4.575   5.5  
education                   0.200   0.25   0.400   3.5  
entertainment               0.300   0.55   1.000   6.8  
food                        0.100   0.20   0.500   2.0  
health                      0.300   0.40   0.500   0.8  
household                   6.525   8.60  13.125  24.7  
miscellenous                7.300   7.70   7.975   9.3  
other                       0.500   0.50   0.675   0.8  
transportation             11.375  13.50  14.950  16.1

1. Highest portion of disposable income spent on Transportation, followed by Housing and household expenses.
2. The median proportion for food is 2% while the mean is 3.6% implying that there is a right skew among income classes.
3. In fact, similar trends follows on all the categories except transportation. this might infer that higher class spend more in every categories.
4. However, this might also be due to the higher number of observations in lower classes.

In [10]:
df.groupby(['income_group', 'Category']).describe()

percentage_of_income             \
                                                            count       mean   
income_group       Category                                                    
Lower Class        alcohol and intoxicants                   48.0   0.604167   
                   clothings and accessories                 90.0   0.421111   
                   communication                              9.0   4.666667   
                   education                                 45.0   0.313333   
                   entertainment                            195.0   0.673846   
                   food                                     297.0   0.440404   
                   health                                    18.0   0.394444   
                   household                                 27.0  11.588889   
                   miscellenous                               9.0   7.288889   
                   other                                      9.0   0.655556   
                   transportation                             9.0  10.811111   
Lower Middle Class alcohol and intoxicants                   64.0   0.554688   
                   clothings and accessories                120.0   0.480833   
                   communication                             12.0   4.191667   
                   education                                 60.0   0.378333   
                   entertainment                            260.0   0.790769   
                   food                                     396.0   0.363889   
                   health                                    24.0   0.400000   
                   household                                 36.0  10.333333   
                   miscellenous                              12.0   7.800000   
                   other                                     12.0   0.541667   
                   transportation                            12.0  13.508333   
Upper Class        alcohol and intoxicants                   16.0   0.475000   
                   clothings and accessories                 30.0   0.603333   
                   communication                              3.0   2.733333   
                   education                                 15.0   0.640000   
                   entertainment                             65.0   0.981538   
                   food                                      99.0   0.245455   
                   health                                     6.0   0.300000   
                   household                                  9.0   9.644444   
                   miscellenous                               3.0   7.733333   
                   other                                      3.0   0.400000   
                   transportation                             3.0  14.466667   
Upper Middle Class alcohol and intoxicants                   32.0   0.534375   
                   clothings and accessories                 60.0   0.570000   
                   communication                              6.0   3.666667   
                   education                                 30.0   0.403333   
                   entertainment                            130.0   0.922308   
                   food                                     198.0   0.303535   
                   health                                    12.0   0.425000   
                   household                                 18.0   9.666667   
                   miscellenous                               6.0   7.916667   
                   other                                      6.0   0.500000   
                   transportation                             6.0  14.983333   

                                                                             \
                                                   std   min     25%    50%   
income_group       Category                                                   
Lower Class        alcohol and intoxicants    0

1. Maximum spent on "household expenses" by Lower and Lower middle class families is more than 20% of their income. While upper class and upper middle class spend half that.
2. The mean and median also follow similar trends; Upper Class and upper Middle class households spend around 3% less in household income than the rest.
3. In fact, proportions spent in household expenses decrease as income increases.
4. 

In [11]:
df.groupby(['Category','income_group']).describe()

percentage_of_income             \
                                                            count       mean   
Category                  income_group                                         
alcohol and intoxicants   Lower Class                        48.0   0.604167   
                          Lower Middle Class                 64.0   0.554688   
                          Upper Class                        16.0   0.475000   
                          Upper Middle Class                 32.0   0.534375   
clothings and accessories Lower Class                        90.0   0.421111   
                          Lower Middle Class                120.0   0.480833   
                          Upper Class                        30.0   0.603333   
                          Upper Middle Class                 60.0   0.570000   
communication             Lower Class                         9.0   4.666667   
                          Lower Middle Class                 12.0   4.191667   
                          Upper Class                         3.0   2.733333   
                          Upper Middle Class                  6.0   3.666667   
education                 Lower Class                        45.0   0.313333   
                          Lower Middle Class                 60.0   0.378333   
                          Upper Class                        15.0   0.640000   
                          Upper Middle Class                 30.0   0.403333   
entertainment             Lower Class                       195.0   0.673846   
                          Lower Middle Class                260.0   0.790769   
                          Upper Class                        65.0   0.981538   
                          Upper Middle Class                130.0   0.922308   
food                      Lower Class                       297.0   0.440404   
                          Lower Middle Class                396.0   0.363889   
                          Upper Class                        99.0   0.245455   
                          Upper Middle Class                198.0   0.303535   
health                    Lower Class                        18.0   0.394444   
                          Lower Middle Class                 24.0   0.400000   
                          Upper Class                         6.0   0.300000   
                          Upper Middle Class                 12.0   0.425000   
household                 Lower Class                        27.0  11.588889   
                          Lower Middle Class                 36.0  10.333333   
                          Upper Class                         9.0   9.644444   
                          Upper Middle Class                 18.0   9.666667   
miscellenous              Lower Class                         9.0   7.288889   
                          Lower Middle Class                 12.0   7.800000   
                          Upper Class                         3.0   7.733333   
                          Upper Middle Class                  6.0   7.916667   
other                     Lower Class                         9.0   0.655556   
                          Lower Middle Class                 12.0   0.541667   
                          Upper Class                         3.0   0.400000   
                          Upper Middle Class                  6.0   0.500000   
transportation            Lower Class                         9.0  10.811111   
                          Lower Middle Class                 12.0  13.508333   
                          Upper Class                         3.0  14.466667   
                          Upper Middle Class                  6.0  14.983333   

                                                                             \
                                                   std   min     25%    50%   
Category                  income_group                                        
alcohol and intoxicants   Lower Class         0

1. Lower class households spend more proportion of their income in alcohol and intoxicants. In terms of actual sum spent is 1.6 for lower class, and 3.5 pounds.
2. There is a gradual increase in proportion spent in clothing and accessories. Lower class spend around 0.42% of their income while upper class spend around 0.6%. This is a expected outcome.
3. There is a interesting trend in communication expenses, while lowerclass spend 4.6% in communication, the expense gradually decreases to 2.7% in upper class. This is due to the TV plans, broadands, mobile plans costing the same for everyone, however disposable income increasing across class. This might be where a rise in price negatively affects lower income households.
4. There is significant difference in income spent in entertainment. Upper class households spend more in travel, holiday, eating out, compared to lower class households. If we compare the flat numbers estimation, the difference doubles up : lower class £1.8, lower middle class £3.08, upper middle class £ 5.75, upper class £7.3 weekly.
5. In terms of food, as expected, lower class spend higher proportion of their income in food than upper class.
6. In terms of health related expenses, lower, lower-middle, and upper-middle class families spend similar proportion of their disposable income, meaning the flat cash spend rises with income. Upper class spend considerably lower than others.
6. In terms of household expenses, the percentage of disposable income spent increases, lower the class level.
7. Miscellenous expenses like personal care items, toiletries have similar proportion spent in all households. This implies, upper class households opt for more expensive options.
8. However, upper class people spend higher proportion of their income in transportation (purchase, fuel, taxes), compared to lower class households.

## Hypothesis Testing

In [6]:
#df2 = df.loc[df['Category'].isin(['clothings and accessories','communication','transportation','miscellenous'])]
df.groupby('Category').corr()

percentage_of_income  \
Category                                                               
alcohol and intoxicants   percentage_of_income              1.000000   
                          avg weekly income                -0.111199   
clothings and accessories percentage_of_income              1.000000   
                          avg weekly income                 0.066728   
communication             percentage_of_income              1.000000   
                          avg weekly income                -0.766152   
education                 percentage_of_income              1.000000   
                          avg weekly income                 0.151614   
entertainment             percentage_of_income              1.000000   
                          avg weekly income                 0.114623   
food                      percentage_of_income              1.000000   
                          avg weekly income                -0.186480   
health                    percentage_of_income              1.000000   
                          avg weekly income                -0.060543   
household                 percentage_of_income              1.000000   
                          avg weekly income                -0.147608   
miscellenous              percentage_of_income              1.000000   
                          avg weekly income                 0.373806   
other                     percentage_of_income              1.000000   
                          avg weekly income                -0.579522   
transportation            percentage_of_income              1.000000   
                          avg weekly income                 0.746328   

                                                avg weekly income  
Category                                                           
alcohol and intoxicants   percentage_of_income          -0.111199  
                          avg weekly income              1.000000  
clothings and accessories percentage_of_income           0.066728  
                          avg weekly income              1.000000  
communication             percentage_of_income          -0.766152  
                          avg weekly income              1.000000  
education                 percentage_of_income           0.151614  
                          avg weekly income              1.000000  
entertainment             percentage_of_income           0.114623  
                          avg weekly income              1.000000  
food                      percentage_of_income          -0.186480  
                          avg weekly income              1.000000  
health                    percentage_of_income          -0.060543  
                          avg weekly income              1.000000  
household                 percentage_of_income          -0.147608  
                          avg weekly income              1.000000  
miscellenous              percentage_of_income           0.373806  
                          avg weekly income              1.000000  
other                     percentage_of_income          -0.579522  
                          avg weekly income              1.000000  
transportation            percentage_of_income           0.746328  
                          avg weekly income              1.000000

## ANOVA and Lavene

In [2]:
import scipy
from scipy import stats

In [7]:
#preparing data for transportarion
dft = df.loc[df['Category'].isin(['transportation'])]


In [10]:
dftl = dft.loc[dft['income_group'].isin(['Lower Class'])] #lowerclass
dftlm = dft.loc[dft['income_group'].isin(['Lower Middle Class'])]
dftum = dft.loc[dft['income_group'].isin(['Upper Middle Class'])]
dftu = dft.loc[dft['income_group'].isin(['Upper Class'])]

In [11]:

scipy.stats.levene(dftl['percentage_of_income'],dftlm['percentage_of_income'],dftum['percentage_of_income'],dftu['percentage_of_income'], center='median', proportiontocut=0.05)

LeveneResult(statistic=0.8531820324957745, pvalue=0.4776260709019463)

### Communication

In [12]:
dft = df.loc[df['Category'].isin(['communication'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class'])] #lowerclass
dftlm = dft.loc[dft['income_group'].isin(['Lower Middle Class'])]
dftum = dft.loc[dft['income_group'].isin(['Upper Middle Class'])]
dftu = dft.loc[dft['income_group'].isin(['Upper Class'])]

In [13]:
scipy.stats.levene(dftl['percentage_of_income'],
                   dftlm['percentage_of_income'],
                   dftum['percentage_of_income'],
                   dftu['percentage_of_income'], center='median', proportiontocut=0.05)

LeveneResult(statistic=0.6242448942625551, pvalue=0.6057538988841897)

### miscellaneous

In [14]:
dft = df.loc[df['Category'].isin(['miscellenous'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class'])] #lowerclass
dftlm = dft.loc[dft['income_group'].isin(['Lower Middle Class'])]
dftum = dft.loc[dft['income_group'].isin(['Upper Middle Class'])]
dftu = dft.loc[dft['income_group'].isin(['Upper Class'])]

In [15]:
scipy.stats.levene(dftl['percentage_of_income'],
                   dftlm['percentage_of_income'],
                   dftum['percentage_of_income'],
                   dftu['percentage_of_income'], center='median', proportiontocut=0.05)

LeveneResult(statistic=1.4854500177872607, pvalue=0.24166887593649805)

In [16]:
dft = df.loc[df['Category'].isin(['clothings and accessories'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class'])] #lowerclass
dftlm = dft.loc[dft['income_group'].isin(['Lower Middle Class'])]
dftum = dft.loc[dft['income_group'].isin(['Upper Middle Class'])]
dftu = dft.loc[dft['income_group'].isin(['Upper Class'])]

In [17]:
scipy.stats.levene(dftl['percentage_of_income'],
                   dftlm['percentage_of_income'],
                   dftum['percentage_of_income'],
                   dftu['percentage_of_income'], center='median', proportiontocut=0.05)

LeveneResult(statistic=0.6271911552629782, pvalue=0.5979432351672107)

## Paired Sample T-Test

### Transportation

In [59]:
dft = df.loc[df['Category'].isin(['transportation'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class','Lower Middle Class'])] #lowerclass
dftl1 = dftl.loc[dftl['year'] == 2019]
dftl2 = dftl.loc[dftl['year'] == 2020]
dftu = dft.loc[dft['income_group'].isin(['Upper Class','Upper Middle Class'])]
dftu1 = dftu.loc[dftu['year'] == 2019]
dftu2 = dftu.loc[dftu['year'] == 2020]

In [42]:
stats.ttest_rel(dftl1['percentage_of_income'], dftl2['percentage_of_income'])

Ttest_relResult(statistic=2.340568492816482, pvalue=0.0577989840349328)

In [60]:
scipy.stats.wilcoxon(dftl1['percentage_of_income'], dftl2['percentage_of_income'])

WilcoxonResult(statistic=3.0, pvalue=0.078125)

In [46]:
stats.ttest_rel(dftu1['percentage_of_income'], dftu2['percentage_of_income'])

Ttest_relResult(statistic=0.2165063509461087, pvalue=0.8486700183084052)

In [61]:
scipy.stats.wilcoxon(dftu1['percentage_of_income'], dftu2['percentage_of_income'])

WilcoxonResult(statistic=2.0, pvalue=0.75)

Paired t-test
1. H4 : null hypothesis of equal averages cannot be rejected for lower income households' spent in transportation before and after pandemic. 
2. H8 : null hypothesis of equal averages cannot be rejected for higher income households' spent in transportation before and after pandemic

Wilcoxon Signed-rank Test
1. H4 : null hypothesis of equal averages cannot be rejected for lower income households' spent in transportation before and after pandemic. 
2. H8 : null hypothesis of equal averages cannot be rejected for higher income households' spent in transportation before and after pandemic

Both the tests show that there was no statistically significant effect of pandemic on income spent on transportation to both upper and lower income households.

### Communication

In [68]:
dft = df.loc[df['Category'].isin(['communication'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class','Lower Middle Class'])] #lowerclass
dftl1 = dftl.loc[dftl['year'] == 2019]
dftl2 = dftl.loc[dftl['year'] == 2020]
dftu = dft.loc[dft['income_group'].isin(['Upper Class','Upper Middle Class'])]
dftu1 = dftu.loc[dftu['year'] == 2019]
dftu2 = dftu.loc[dftu['year'] == 2020]

In [69]:
stats.ttest_rel(dftl1['percentage_of_income'], dftl2['percentage_of_income'])


Ttest_relResult(statistic=-0.29488391230979394, pvalue=0.7780198253944598)

In [71]:
scipy.stats.wilcoxon(dftl1['percentage_of_income'], dftl2['percentage_of_income'])

WilcoxonResult(statistic=9.0, pvalue=0.7518296340458492)

In [72]:
stats.ttest_rel(dftu1['percentage_of_income'], dftu2['percentage_of_income'])

Ttest_relResult(statistic=0.3779644730092272, pvalue=0.7418011102528389)

In [73]:
scipy.stats.wilcoxon(dftu1['percentage_of_income'], dftu2['percentage_of_income'])

WilcoxonResult(statistic=1.0, pvalue=0.6547208460185769)

Paired t-test
1. H5 : null hypothesis of equal averages cannot be rejected for lower income households' spent in Communication before and after pandemic. 
2. H9 : null hypothesis of equal averages cannot be rejected for higher income households' spent in Communication before and after pandemic

Wilcoxon Signed-rank test

1. H5 : null hypothesis of equal averages cannot be rejected for lower income households' spent in Communication before and after pandemic. 
2. H9 : null hypothesis of equal averages cannot be rejected for higher income households' spent in Communication before and after pandemic

There was no statistically significant effect of pandemic on income spent on Communication to both upper and lower income households.

### Miscellaneous

In [79]:
dft = df.loc[df['Category'].isin(['miscellenous'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class','Lower Middle Class'])] #lowerclass
dftl1 = dftl.loc[dftl['year'] == 2019]
dftl2 = dftl.loc[dftl['year'] == 2020]
dftu = dft.loc[dft['income_group'].isin(['Upper Class','Upper Middle Class'])]
dftu1 = dftu.loc[dftu['year'] == 2019]
dftu2 = dftu.loc[dftu['year'] == 2020]

In [80]:
stats.ttest_rel(dftl1['percentage_of_income'], dftl2['percentage_of_income'])

Ttest_relResult(statistic=0.24540954937812107, pvalue=0.8143178852439519)

In [81]:
scipy.stats.wilcoxon(dftl1['percentage_of_income'], dftl2['percentage_of_income'])

WilcoxonResult(statistic=12.0, pvalue=0.8125)

In [82]:
stats.ttest_rel(dftu1['percentage_of_income'], dftu2['percentage_of_income'])

Ttest_relResult(statistic=-0.6401843996644789, pvalue=0.5876069505788393)

In [83]:
scipy.stats.wilcoxon(dftu1['percentage_of_income'], dftu2['percentage_of_income'])

WilcoxonResult(statistic=2.0, pvalue=0.75)

for both paired t-test and Wilcoxon Signed-rank test
1. H6 : null hypothesis of equal averages cannot be rejected for lower income households' spent in miscellaneous before and after pandemic. 
2. H10 : null hypothesis of equal averages cannot be rejected for higher income households' spent in miscellaneous before and after pandemic

There was no statistically significant effect of pandemic on income spent on miscellaneous to both upper and lower income households. 

In [84]:
dft = df.loc[df['Category'].isin(['clothings and accessories'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class','Lower Middle Class'])] #lowerclass
dftl1 = dftl.loc[dftl['year'] == 2019]
dftl2 = dftl.loc[dftl['year'] == 2020]
dftu = dft.loc[dft['income_group'].isin(['Upper Class','Upper Middle Class'])]
dftu1 = dftu.loc[dftu['year'] == 2019]
dftu2 = dftu.loc[dftu['year'] == 2020]

In [85]:
stats.ttest_rel(dftl1['percentage_of_income'], dftl2['percentage_of_income'])

Ttest_relResult(statistic=-2.549611272245946, pvalue=0.013009901652070315)

In [86]:
scipy.stats.wilcoxon(dftl1['percentage_of_income'], dftl2['percentage_of_income'])

WilcoxonResult(statistic=177.0, pvalue=0.10286361845355033)

In [55]:
stats.ttest_rel(dftu1['percentage_of_income'], dftu2['percentage_of_income'])

Ttest_relResult(statistic=-1.772664126797008, pvalue=0.0867902585479985)

In [87]:
scipy.stats.wilcoxon(dftu1['percentage_of_income'], dftu2['percentage_of_income'])

WilcoxonResult(statistic=35.5, pvalue=0.1591966277971013)

Paired t-test
1. H7 : null hypothesis of equal averages is rejected for lower income households' spent in clothings and accessories before and after pandemic. 
2. H11 : null hypothesis of equal averages cannot be rejected for higher income households' spent in clothings and accessories before and after pandemic

Wilcoxon signed-rank test

1. H7 : null hypothesis of equal averages could not be rejected for lower income households' spent in clothings and accessories before and after pandemic. 
2. H11: null hypothesis of equal averages cannot be rejected for higher income households' spent in clothings and accessories before and after pandemic

According to the t - test, there was statistically significant effect of pandemic on income spent on clothings and accessories to lower income households but not in upper income households. However, wilcoxon signed rank test signify that pandemic has no statistically significant effect.

### covid vs no covid

## Transportation

In [8]:
dft = df.loc[df['Category'].isin(['transportation'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class','Lower Middle Class'])] #lowerclass
dftl1 = dftl.loc[dftl['year'] == 2019]
dftl2 = dftl.loc[dftl['year'] == 2021]
dftu = dft.loc[dft['income_group'].isin(['Upper Class','Upper Middle Class'])]
dftu1 = dftu.loc[dftu['year'] == 2019]
dftu2 = dftu.loc[dftu['year'] == 2021]

In [9]:
print(stats.ttest_rel(dftl1['percentage_of_income'], dftl2['percentage_of_income']))
print(scipy.stats.wilcoxon(dftl1['percentage_of_income'], dftl2['percentage_of_income']))
print(stats.ttest_rel(dftu1['percentage_of_income'], dftu2['percentage_of_income']))
print(scipy.stats.wilcoxon(dftu1['percentage_of_income'], dftu2['percentage_of_income']))


Ttest_relResult(statistic=4.0120825499016775, pvalue=0.0070217478497620915)
WilcoxonResult(statistic=1.0, pvalue=0.03125)
Ttest_relResult(statistic=6.285714285714281, pvalue=0.024387845651315722)
WilcoxonResult(statistic=0.0, pvalue=0.25)


### Communication

In [11]:
dft = df.loc[df['Category'].isin(['communication'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class','Lower Middle Class'])] #lowerclass
dftl1 = dftl.loc[dftl['year'] == 2019]
dftl2 = dftl.loc[dftl['year'] == 2021]
dftu = dft.loc[dft['income_group'].isin(['Upper Class','Upper Middle Class'])]
dftu1 = dftu.loc[dftu['year'] == 2019]
dftu2 = dftu.loc[dftu['year'] == 2021]

In [12]:
print(stats.ttest_rel(dftl1['percentage_of_income'], dftl2['percentage_of_income']))
print(scipy.stats.wilcoxon(dftl1['percentage_of_income'], dftl2['percentage_of_income']))
print(stats.ttest_rel(dftu1['percentage_of_income'], dftu2['percentage_of_income']))
print(scipy.stats.wilcoxon(dftu1['percentage_of_income'], dftu2['percentage_of_income']))

Ttest_relResult(statistic=-5.908789478687512, pvalue=0.0010451022906022807)
WilcoxonResult(statistic=0.0, pvalue=0.015625)
Ttest_relResult(statistic=-6.928203230275516, pvalue=0.02020410288672872)
WilcoxonResult(statistic=0.0, pvalue=0.25)


### Miscellaneous

In [13]:
dft = df.loc[df['Category'].isin(['miscellenous'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class','Lower Middle Class'])] #lowerclass
dftl1 = dftl.loc[dftl['year'] == 2019]
dftl2 = dftl.loc[dftl['year'] == 2021]
dftu = dft.loc[dft['income_group'].isin(['Upper Class','Upper Middle Class'])]
dftu1 = dftu.loc[dftu['year'] == 2019]
dftu2 = dftu.loc[dftu['year'] == 2021]
print(stats.ttest_rel(dftl1['percentage_of_income'], dftl2['percentage_of_income']))
print(scipy.stats.wilcoxon(dftl1['percentage_of_income'], dftl2['percentage_of_income']))
print(stats.ttest_rel(dftu1['percentage_of_income'], dftu2['percentage_of_income']))
print(scipy.stats.wilcoxon(dftu1['percentage_of_income'], dftu2['percentage_of_income']))

Ttest_relResult(statistic=-0.39931042923725046, pvalue=0.7034897016877237)
WilcoxonResult(statistic=11.0, pvalue=0.6875)
Ttest_relResult(statistic=2.9605947323337434e-15, pvalue=0.9999999999999979)
WilcoxonResult(statistic=3.0, pvalue=1.0)


### Clothing and Accessories

In [14]:
dft = df.loc[df['Category'].isin(['clothings and accessories'])]
dftl = dft.loc[dft['income_group'].isin(['Lower Class','Lower Middle Class'])] #lowerclass
dftl1 = dftl.loc[dftl['year'] == 2019]
dftl2 = dftl.loc[dftl['year'] == 2021]
dftu = dft.loc[dft['income_group'].isin(['Upper Class','Upper Middle Class'])]
dftu1 = dftu.loc[dftu['year'] == 2019]
dftu2 = dftu.loc[dftu['year'] == 2021]
print(stats.ttest_rel(dftl1['percentage_of_income'], dftl2['percentage_of_income']))
print(scipy.stats.wilcoxon(dftl1['percentage_of_income'], dftl2['percentage_of_income']))
print(stats.ttest_rel(dftu1['percentage_of_income'], dftu2['percentage_of_income']))
print(scipy.stats.wilcoxon(dftu1['percentage_of_income'], dftu2['percentage_of_income']))

Ttest_relResult(statistic=5.191055919091155, pvalue=2.0064088740500793e-06)
WilcoxonResult(statistic=109.5, pvalue=5.993755450428454e-06)
Ttest_relResult(statistic=2.8676042897192255, pvalue=0.007630149582617086)
WilcoxonResult(statistic=30.0, pvalue=0.008259785249635538)
